src/Core/Scheduler/Providers/TMS_Provider.js

```diff
@@ -0,0 +1,69 @@
+import * as THREE from 'three';
+import OGCWebServiceHelper from './OGCWebServiceHelper';
+import Extent from '../../Geographic/Extent';
+
+function TMS_Provider() {
+}
+
+TMS_Provider.prototype.preprocessDataLayer = function preprocessDataLayer(layer) {
+    if (!layer.extent) {
+        throw new Error(`Missing extent property for layer '${layer.id}'`);
+    }
+    if (!layer.projection) {
+        throw new Error(`Missing projection property for layer '${layer.id}'`);
+    }
+    layer.extent = new Extent(layer.projection, ...layer.extent);
+    if (!layer.options.zoom) {
+        layer.options.zoom = {
+            min: 0,
+            max: 18,
```

Comment:
`getCapabilities`... I forgot about this one. I need to resume the work on that.